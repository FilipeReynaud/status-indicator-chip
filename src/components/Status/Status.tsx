import React from "react";

// Statuses
import Active from "../Active";
import Inactive from "../Inactive";

export interface InactiveProps {
	type: "active" | "inactive";
}

const StatusChip = (props: InactiveProps) => {
	const { type } = props;

	if (type === "active") {
		return <Active />;
	} else if (type === "inactive") {
		return <Inactive />;
	} else {
		return null;
	}
};

export default StatusChip;
