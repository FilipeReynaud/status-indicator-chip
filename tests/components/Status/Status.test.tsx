import React from "react";
import { render } from "@testing-library/react";

// Components
import Status from "../../../src/components/Status";

// Types
import { StatusChipProps } from "../../../src/components/Status/Status";

describe("Status", () => {
	const defaultTestProps = {
		type: "active",
		fontColor: "blue",
		backgroundColor: "lightblue",
		label: "Test",
		withBlinkAnimation: true,
		size: "md",
	} as StatusChipProps;

	it("active status chip should be rendered properly", () => {
		// Act
		const { container } = render(<Status type="active" />);

		// Assert
		expect(container).toMatchSnapshot();
	});

	it("inactive status chip should be rendered properly", () => {
		// Act
		const { container } = render(<Status type="inactive" />);

		// Assert
		expect(container).toMatchSnapshot();
	});

	it("custom status chip should be rendered properly", () => {
		// Act
		const testProps = {
			...defaultTestProps,
			type: "custom",
		} as StatusChipProps;
		const { container } = render(<Status {...testProps} />);

		// Assert
		expect(container).toMatchSnapshot();
	});
});
