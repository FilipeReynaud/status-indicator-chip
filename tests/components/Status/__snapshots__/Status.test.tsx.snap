// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status active status chip should be rendered properly 1`] = `
<div>
  <table
    class="status-indicator-table"
    data-test="status-indicator-table"
    style="padding: 0px 7px 0px 5px; background-color: rgb(221, 236, 209);"
  >
    <tbody>
      <tr>
        <td>
          <div
            class="status-indicator-circle"
            data-test="status-indicator-circle"
            style="width: 10px; height: 10px; background-color: rgb(14, 175, 0); --animation-color: #0eaf00; animation-name: blink; animation-duration: 1s; animation-iteration-count: infinite;"
          />
        </td>
        <td>
          <span
            class="status-indicator-label"
            data-test="status-indicator-label"
            style="color: rgb(14, 175, 0); font-size: 11px;"
          >
            Active
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Status custom status chip should be rendered properly 1`] = `
<div>
  <table
    class="status-indicator-table"
    data-test="status-indicator-table"
    style="padding: 0px 7px 0px 5px; background-color: lightblue;"
  >
    <tbody>
      <tr>
        <td>
          <div
            class="status-indicator-circle"
            data-test="status-indicator-circle"
            style="width: 10px; height: 10px; background-color: blue; --animation-color: blue; animation-name: blink; animation-duration: 1s; animation-iteration-count: infinite;"
          />
        </td>
        <td>
          <span
            class="status-indicator-label"
            data-test="status-indicator-label"
            style="color: blue; font-size: 11px;"
          >
            Test
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Status inactive status chip should be rendered properly 1`] = `
<div>
  <table
    class="status-indicator-table"
    data-test="status-indicator-table"
    style="padding: 0px 7px 0px 5px; background-color: rgb(253, 238, 238);"
  >
    <tbody>
      <tr>
        <td>
          <div
            class="status-indicator-circle"
            data-test="status-indicator-circle"
            style="width: 10px; height: 10px; background-color: rgb(238, 95, 91);"
          />
        </td>
        <td>
          <span
            class="status-indicator-label"
            data-test="status-indicator-label"
            style="color: rgb(238, 95, 91); font-size: 11px;"
          >
            Inactive
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
